### DISCUSSION TITLE
npm lock file maintenance stopped working

### DISCUSSION THREAD

**Discussion by dominykas** [renovatebot/renovate#25030] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Minimal reproduction case repository: 

- https://github.com/dominykas/renovate-lock-file-maintenance/blob/main/renovate.json
- https://github.com/dominykas/renovate-lock-file-maintenance/blob/main/package-lock.json (lodash is 4.17.20 in the `package-lock.json`, but a 4.17.21 exists; a PR to upgrade just lodash [does get created](https://github.com/dominykas/renovate-lock-file-maintenance/pull/3) when `"rangeStrategy": "replace"` is not set)

I would expect a "Lock file maintenance" PR, however one does not get created.

This regressed between renovate `v36.99.0` and and `v37.1.3` (judging from our self-hosted setup).

### Relevant debug logs

From https://developer.mend.io/github/dominykas/renovate-lock-file-maintenance/-/job/695e611f-a989-4e88-9ba7-662b0684eea1

<details><summary>Logs</summary>

```
DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=false (branch="renovate/lock-file-maintenance")
DEBUG: No package files need updating (branch="renovate/lock-file-maintenance")
DEBUG: Getting updated lock files (branch="renovate/lock-file-maintenance")
DEBUG: Writing package.json files (branch="renovate/lock-file-maintenance")
{
  "packageFiles": [
    "package.json"
  ]
}

DEBUG: Ensuring package-lock.json is removed (branch="renovate/lock-file-maintenance")
DEBUG: Writing any updated package files (branch="renovate/lock-file-maintenance")
DEBUG: No updated lock files in branch (branch="renovate/lock-file-maintenance")
DEBUG: No files to commit (branch="renovate/lock-file-maintenance")
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
Thanks for reporting! Fix in #25045

**Comment by dominykas** [comment]
Thanks, this works!
